compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/parser/BLangNodeBuilder.java

```diff
@@ -2906,6 +2909,44 @@ private void initializeBLangVariable(BLangVariable var, TypeDescriptorNode type,
         }
     }
 
+    private String getVariableName(BLangVariable variable) {
+        NodeKind kind = variable.getKind();
+        String name = DOLLAR;
+        switch (kind) {
+            case VARIABLE:
+                return name.concat(((BLangSimpleVariable) variable).name.value + DOLLAR);
+            case RECORD_VARIABLE:
+                for (BLangRecordVariableKeyValue keyValue : ((BLangRecordVariable) variable).getVariables()) {
+                    name = name.concat(keyValue.key.value + DOLLAR);
+                }
+                return name;
+            case TUPLE_VARIABLE:
+                for (BLangVariable var : ((BLangTupleVariable) variable).memberVariables) {
+                    name = name.concat(getVariableName(var) + DOLLAR);
+                }
+                return name;
+            case ERROR_VARIABLE:
```

Comment:
I prefer to have these under ERROR_VARIABLE and default case to check some assert to imply that they are handled?

Reply:
We r avoiding asserts asses as much as possible right? Since the cast at L#2930 will fail for any other case I think we r good WDYT?

Reply:
Also, any failures should be handled before this point right? I also this it is okay to have it like this